src/contexts/tokens.context.tsx

```diff
@@ -158,78 +158,90 @@ const TokensProvider: FC<PropsWithChildren> = (props) => {
-        return getNativeTokenInfo({ chain: nativeChain, token })
-          .then(({ originNetwork, originTokenAddress }) => {
-            // if this is the case we use originTokenAddress as native and token.address as wrapped
-            const originalTokenChain = env?.chains.find(
-              (chain) => chain.networkId === originNetwork
-            );
-            if (originalTokenChain === undefined) {
-              throw Error(`Could not find a chain that matched the originNetwork ${originNetwork}`);
-            } else {
-              const newToken: Token = {
-                ...token,
-                address: originTokenAddress,
-                chainId: originalTokenChain.chainId,
-                wrappedToken: {
-                  address: token.address,
-                  chainId: nativeChain.chainId,
-                },
-              };
-              return newToken;
-            }
+        return computeWrappedTokenAddress({
+          nativeChain,
+          otherChain: wrappedChain,
+          token,
+        })
+          .then((wrappedAddress) => {
+            const newToken: Token = {
+              ...token,
+              wrappedToken: {
+                address: wrappedAddress,
+                chainId: wrappedChain.chainId,
+              },
+            };
+            return newToken;
           })
-          .catch(() => {
-            // if the provided address is native we compute the wrapped address
-            return computeWrappedTokenAddress({
-              nativeChain,
-              otherChain: wrappedChain,
-              token,
-            })
-              .then((wrappedAddress) => {
-                const newToken: Token = {
-                  ...token,
-                  wrappedToken: {
-                    address: wrappedAddress,
-                    chainId: wrappedChain.chainId,
-                  },
-                };
-                return newToken;
-              })
-              .catch((e) => {
-                notifyError(e);
-                return Promise.resolve(token);
-              });
+          .catch((e) => {
+            notifyError(e);
+            return Promise.resolve(token);
           });
       }
     },
-    [env, getNativeTokenInfo, computeWrappedTokenAddress, notifyError]
+    [env, computeWrappedTokenAddress, notifyError]
   );
 
   const getTokenFromAddress = useCallback(
-    async ({ address, chain }: GetTokenFromAddressParams): Promise<Token> => {
-      const erc20Contract = Erc20__factory.connect(address, chain.provider);
+    async ({ address }: GetTokenFromAddressParams): Promise<Token> => {
+      if (!env) {
+        throw Error("The env is not ready");
+      }
+
+      if (!isAsyncTaskDataAvailable(connectedProvider)) {
+        throw Error("The connectedProvider is not ready");
+      }
+
+      const chain = env.chains.find((chain) => chain.chainId === connectedProvider.data.chainId);
```

Comment:
Didn't we agree on use the `provider` that was currently selected in the form? The `connectedProvider` could be different than the one selected in the form.

Reply:
mmmm.... what I understood is that we were going to use the provider of the selected network in metamask, not in the form.

Reply:
Take into account that when you are in the form and you search for a token, you'd like to search for it in the network currently selected in the form. If not, if the connected provider is connected to the other network and the token doesn't exist in the network currently selected in the form you'll see an error saying that the token doesn't exist when you expect it to exist.

Reply:
Ok, I will change this, but my reasoning was the same, but for the globally selected network: When you search for a token, you'd like to search for it in the network currently selected in MetaMask.

Having 2 selected networks is a clear design flaw in this application. Some users might expect the behavior you describe, but I'm sure some users would expect the behavior I describe.

Reply:
Well, I don't think so. Usually, when you are in a dapp playing around usually you forget about the network selected in MetaMask very easily. And if the UI has one chain selected, you are going to think that you are connected to that network, so you expect to search the token in that chain. As an example, this is what Jesús was doing the other that.

Reply:
Done